lib/pico.php

```diff
@@ -309,7 +309,7 @@ protected function base_url()
 	protected function get_protocol()
 	{
 		$protocol = 'http';
-		if(isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] != 'off'){
```

Comment:
I agree with @auyongcheemeng, this is the best way to test for HTTP/S

```php
if (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] != 'off') { 
     //HTTPS 
}
```

maybe throw in a `|| $_SERVER['SERVER_PORT'] == 443` to be super-sure..